ygot/pathstrings_test.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ygot
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// TestPathToString validates the functionality provided by the PathToString
+// method of the library.
+func TestPathToString(t *testing.T) {
+	tests := []struct {
+		name    string
+		in      *gnmipb.Path
+		want    string
+		wantErr string
+	}{{
+		name: "root path",
+		in:   &gnmipb.Path{Element: []string{}},
+		want: "/",
+	}, {
+		name: "simple path parts",
+		in:   &gnmipb.Path{Element: []string{"a", "b", "c", "d"}},
+		want: "/a/b/c/d",
+	}, {
```

Comment:
how about no leading /, or trailing /? also empty string.

Reply:
No leading `/` is covered in "relative path" above.

Reply:
Added trailing /, added empty string.